templates/terraform/modules/environment/main.tf

```diff
@@ -2,25 +2,35 @@
 
 locals {
   kubernetes_cluster_name = "${var.project}-${var.environment}-${var.region}"
-}
 
-data "aws_iam_user" "ci_user" {
-  user_name = "${var.project}-ci-user" # Should have been created in the bootstrap process
-}
-
-locals {
-  role_name_list = var.roles.*.name
   users = [
     for u in var.user_role_mapping : {
       name = u.name
       roles = [
         for r in u.roles :
-        r.name if contains(local.role_name_list, r.name) && contains(r.environments, var.environment)
+        r.name if contains(var.roles.*.name, r.name) && contains(r.environments, var.environment)
       ]
     }
   ]
+
+  ci_users = [
+    for u in local.users: u.name if contains(u.roles, "deployer")
```

Comment:
Let's not do it based on if there is "deployer" in the name. That's not at all obvious to people who are using this. Either let's target this specific user, or let's provide some sort of data in the user list that lets us know this is a special type  of user, if we actually need to treat them differently.